---
title: "Analysis of Longevity from Muscle dTORC1 Effector Flies"
author: " Dave Bridges, Isabelle Hatfield and Erika Yates"
date: "February 9, 2015"
output:
  html_document:
    keep_md: yes
---

```{r global_options, include=FALSE}
library(knitr)
opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE,dev=c('png','pdf'))
options(scipen = 9, digits = 5)

```

These studies evaluate the effects of knockdown of several dTORC1 effectors, including Autophagy genes (*Atg5*, *Atg8a* and *Atg8b*) and the SREBP1 ortholog *Hlh106*  This data is located in `r getwd()` and was most recently updated on `r date()`.

## Experimental Design

```{r data-input, echo=FALSE}
require(RCurl, quietly=T)
require(lubridate, quietly=T)
birth_worksheet <- 'https://docs.google.com/spreadsheet/pub?key=0Am2qM2oZ12_jdHhNTjNuc25zdDJKZzR3MzVuUUU4dEE&single=true&gid=0&output=csv' 
death_worksheet <- 'https://docs.google.com/spreadsheet/pub?key=0Aitfmor6rCjYdG9GUlgtRXRuc01EVFJqdmp3WVU4b0E&single=true&gid=0&output=csv'
birth_filename <- '../Data/SREBP-Atg Births Worksheet.csv'
death_filename <- '../Data/SREBP-Atg Death Worksheet.csv'

mapping_data <- read.csv('../Data/gene_mapping.csv')
#this contains the data with the control flies
lifespan_data_file_name <- '../Data/Death Worksheet Summer 2014.csv'
lifespan.data <- read.csv(lifespan_data_file_name)
#get data from what crosses are what
lifespan_data_file_name_births <- '../Data/mhc_birth_worksheet.csv'
lifespan.data.births <- read.csv(lifespan_data_file_name_births )
lifespan.data.births$Total <- rowSums(lifespan.data.births[,10:14], na.rm=T)
all.flies <- sum(lifespan.data.births$Total, na.rm=T)

lifespan.data.births$Cross <- as.factor(paste(lifespan.data.births$Male, lifespan.data.births$Female, sep=" x "))
lifespan.data.births$Driver <- as.factor(sapply(strsplit(as.character(lifespan.data.births$Female), "/"),"[", 1))
#releveled UAS showing TSC1 then Raptor
lifespan.data.births$UAS <- as.factor(sapply(strsplit(as.character(lifespan.data.births$Male), "/"),"[", 1))
#remove analyses where incomplete data is presented
lifespan.data.births.complete <- lifespan.data.births[!(is.na(lifespan.data.births$Total)),]
#merged with gene mapping file
annotated.data <- merge(mapping_data, lifespan.data.births.complete, by="UAS", all.y=T)
annotations <- annotated.data[,c('UAS','Gene','Driver','CrossName')]
annotations <- unique(annotations[c('UAS','Gene','Driver','CrossName')])

annotated.lifespan.data <- merge(lifespan.data, annotations, by='CrossName')
#set UAS as a factor
annotated.lifespan.data$UAS <- as.factor(annotated.lifespan.data$UAS)
annotated.lifespan.data$Birth.Date <- mdy(annotated.lifespan.data$Birth.Date)
annotated.lifespan.data$Death.Date <- mdy(annotated.lifespan.data$Death.Date)
annotated.lifespan.data$age <- annotated.lifespan.data$Death.Date - annotated.lifespan.data$Birth.Date
annotated.lifespan.data <- droplevels(annotated.lifespan.data)
annotated.lifespan.data$UAS <- relevel(annotated.lifespan.data$UAS, ref='36304')


#extract only the data that corresponds to the control flies
control.data <- subset(annotated.lifespan.data, Gene=="Control")
#reformat age column
control.data$Age <- as.integer(control.data$age)

birth_data <- read.csv(birth_filename)
birth_data$CrossName <- tolower(birth_data$CrossName)
death_data <- read.csv(death_filename)
death_data$CrossName <- tolower(death_data$CrossName)

effector.data <- merge(death_data, birth_data[duplicated(birth_data$CrossName)==FALSE,c('CrossName','Generation','Cross','Replicate','DateStarted', 'Male', 'Female')], by='CrossName')
effector.data$Birth.Date <- mdy(gsub("/", "-", effector.data$Birth.Date))
effector.data$Death.Date <- mdy(gsub("/", "-", effector.data$Death.Date))
effector.data$Age <- as.integer(effector.data$Death.Date - effector.data$Birth.Date)
effector.data$Cross <- as.factor(paste(effector.data$Male, effector.data$Female, sep=" x "))

#this is complicated in this group
effector.data$Driver <- as.factor(sapply(strsplit(as.character(effector.data$Female), "/"),"[", 1))
effector.data$UAS <- as.factor(effector.data$Male)

effector.data <- merge(effector.data, mapping_data)
#moved gender to being case insensitive
effector.data$Gender <- as.factor(tolower(effector.data$Gender))
#made genotype case insensitive
effector.data$Phenotype <- as.factor(tolower(effector.data$Phenotype))
effector.data$Genotype[effector.data$Phenotype =='wt'] <- "shRNA/GAL4"
effector.data$Genotype[effector.data$Phenotype == 'sb'] <- "shRNA/Tm3"
#made genotype into a factor
effector.data$Genotype <- as.factor(effector.data$Genotype)
#set levels of genotype
effector.data$Genotype <- factor(effector.data$Genotype, levels=c("shRNA/GAL4","shRNA/Tm3"))
effector.data$Genotype <- relevel(effector.data$Genotype, ref="shRNA/GAL4")

effector.data <- droplevels(effector.data)
library(dplyr)
combined.data <- rbind(select(effector.data, Gender, Gene, UAS, Driver, Age),
                       select(control.data, Gender, Gene, UAS, Driver, Age))
write.csv(combined.data, "../Data/Effector Longevity Analysis Data.csv")
```

This script was most recently run on `r date()`.  There has been a total of **`r dim(death_data)[1]`** deaths, with **`r dim(death_data[death_data$Accidental==FALSE,])[1]`** of natural causes and **`r dim(death_data[death_data$Phenotype!='?'&death_data$Accidental==FALSE,])[1]`** of identifiable genotypes excluding accidental deaths.  The oldest fly recorded so far was `r max(combined.data$Age, na.rm=T)` days old at time of death.

## Gene Level Analysis

```{r summary-statistics}
require(plyr)
require(reshape2)

complete.data <- combined.data[!(is.na(combined.data$Age)),]

summarised.data.all <- ddply(complete.data, .(Driver,UAS,Genotype), summarise,
                         Deaths = length(Age),
                         Median.Age = median(Age),
                         Age= mean(Age),
                         Error = sd(Age, na.rm=T)/sqrt(length(Age)))
summarised.data.natural <- ddply(data[data$Accidental==FALSE,], .(Driver,UAS,Genotype), summarise,
                         Deaths = length(Age),
                         Median.Age = median(Age),
                         Age= mean(Age),
                         Error = sd(Age, na.rm=T)/sqrt(length(Age)))

summarised.data.natural.genes <- ddply(complete.data[complete.data$Accidental==FALSE&complete.data$Genotype!='shRNA/GAL4',], .(UAS,Driver,Gene,Genotype), summarise,
                         Deaths = length(Age),
                         Median.Age = median(Age, na.rm=T),
                         Age= mean(Age, na.rm=T),
                         Error = sd(Age, na.rm=T))

summarised.data.natural.gene <- ddply(data[data$Accidental==FALSE,], .(Driver,Gene,Genotype), summarise,
                         Deaths = length(Age),
                         Median.Age = median(Age),
                         Age= mean(Age),
                         Error = sd(Age, na.rm=T)/sqrt(length(Age)))
summarised.data.natural.gene.gender <- ddply(data[data$Accidental==FALSE,], .(Driver,Gene,Genotype,Gender), summarise,
                         Deaths = length(Age),
                         Median.Age = median(Age),
                         Age= mean(Age),
                         Error = sd(Age, na.rm=T)/sqrt(length(Age)))
```



## Number of Flies Examined

The total number of deaths for each cross and genotype, removing deaths that were accidental or not due to natural causes, the data is shown in the table below:

```{r selected-data}
require(xtable)
kable(dcast(summarised.data.natural.gene.gender, Driver+Gene+Gender~Genotype, value.var='Deaths', margins=F, sum), caption = "Total Natural Deaths by Gender for Each Gene and Genotype", label = "tab:analysed-deaths-gene-gender")
```  

The distribution of deaths in the samples are shown at a gender level in the Figure below:

```{r histogram-gender}
with(data[data$Accidental==FALSE&data$Gender=='male'&data$Genotype!='Tm3/Tm6B'&data$Genotype!='NA',], plot(density(Age, na.rm=T), col="black", las=1))
with(data[data$Accidental==FALSE&data$Gender=='female'&data$Genotype!='Tm3/Tm6B'&data$Genotype!='NA',], lines(density(Age, na.rm=T), main="Natural Death Distribution", col="red", las=1, xlab=""))
legend("topright", c("Male","Female"), lty=1, col=c("black","red"), bty="n")
```


## Survival Analysis

All of these are relative to the reference Genotype which is the knockdown (GAL4/shRNA).  The key packages used in this The summary statistics from this analysis are shown in the tables below:

```{r survival-analysis-24b}
require(RColorBrewer)
genotype.colors <- brewer.pal(2, "Set2")
require(survival)
summary.coxph.24b.gene <- data.frame(row.names=levels(data$Gene), 
                                n=rep(NA,length(levels(data$Gene))), 
                                logtest.p=rep(NA,length(levels(data$Gene))),
                                waldtest.p=rep(NA,length(levels(data$Gene))),
                                sctest.p=rep(NA,length(levels(data$Gene))))
for (gene in levels(data$Gene)[1:4]) {
    data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene==gene&data$Genotype!='Tm3/Tm6B',])
    survobj <- with(data.used, Surv(Age, Accidental==FALSE))
    survfit <- survfit(survobj~Genotype, data=data.used)
    #print(sprintf('Analysis of %s with 24B-GAL4 Driver', gene))
    #print(survfit(survobj~Genotype, data=data.used), print.rmean=TRUE)
    coxph.fit <- coxph(survobj~Genotype, data=data.used)
    beta <- coef(coxph.fit)
    se <- sqrt(diag(coxph.fit$var))
    assign(paste('coxph.table.24b',gene,sep='.'),data.frame(Coef=beta, 
                              SE = se,
                              Hazard.Ratio = 1/exp(beta),
                              p = 1 - pchisq((beta/se)^2, 1)))
    
    summary.coxph.24b.gene[gene,] <- c(
      n=coxph(survobj~Genotype, data=data.used)$nevent,
      logtest.p=summary(coxph(survobj~Genotype, data=data.used))$logtest[3],
      waldtest.p=summary(coxph(survobj~Genotype, data=data.used))$wald[3],
      sctest.p=summary(coxph(survobj~Genotype, data=data.used))$sctest[3])
    #summary.coxph.24b$padj <- p.adjust(summary.coxph.24b$logrank.p,n=3,method="BH")
    
    pdf(sprintf('figure/24B-%s-survival-plot.pdf', gene))
    plot(survfit, main = sprintf('Survival of %s with 24B-GAL4 Driver', gene),
         xlab= 'Days',
         ylab= 'Percent Survival',
         yscale=100, 
         las = 1,
         lty=c(1,2,2),
         col= genotype.colors)
    legend("bottomleft", levels(data.used$Genotype), col=genotype.colors, lty=c(1,2,2), bty="n")
    dev.off()
}


#Separating out genders for this analysis
summary.coxph.24b.gene.gender <- data.frame(
                row.names=paste(c("Atg5","Atg5","Atg8a","Atg8a","Atg8b","Atg8b","HLH106","HLH106"),c(rep(c("male","female"),4))),
                n=rep(NA,8), 
                logtest.p=rep(NA,8),
                waldtest.p=rep(NA,8),
                sctest.p=rep(NA,8))

#iterate over gender, then over each gene
for (gender in levels(data$Gender)) {
    for (gene in levels(data$Gene)[1:4]) {
        data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene==gene&data$Genotype!='Tm3/Tm6B'&data$Gender==gender,])
        survobj <- with(data.used, Surv(Age, Accidental==FALSE))
        survfit <- survfit(survobj~Genotype, data=data.used)
        #print(survfit(survobj~Genotype, data=data.used), print.rmean=TRUE)
        #print(coxph(survobj~Genotype, data=data.used))
        
        coxph.fit <- coxph(survobj~Genotype, data=data.used)
        beta <- coef(coxph.fit)
        se <- sqrt(diag(coxph.fit$var))
        assign(paste(paste('coxph.table.24b',gene,sep='.'),gender, sep="."),
               data.frame(Coef=beta, 
                              SE = se,
                              Hazard.Ratio = 1/exp(beta),
                              p = 1 - pchisq((beta/se)^2, 1)))
        
        summary.coxph.24b.gene.gender[paste(gene, gender),] <- c(
          n=coxph(survobj~Genotype, data=data.used)$nevent,
          logtest.p=summary(coxph(survobj~Genotype, data=data.used))$logtest[3],
          waldtest.p=summary(coxph(survobj~Genotype, data=data.used))$wald[3],
          sctest.p=summary(coxph(survobj~Genotype, data=data.used))$sctest[3])
        #summary.coxph.24b$padj <- p.adjust(summary.coxph.24b$logrank.p,n=3,method="BH")
        pdf(sprintf('figure/24B-%s-survival-plot.pdf', paste(gene,gender,sep="-")))
        plot(survfit, main = sprintf('Survival of %s with 24B-GAL4 Driver', paste(gene,gender)),
             xlab= 'Days',
             ylab= 'Percent Survival',
             yscale=100, 
             las = 1,
             lty=c(1,2,2,2),
             col= genotype.colors)
        legend("bottomleft", levels(data$Genotype), col=genotype.colors, lty=c(1,2,2,2), bty="n")
        dev.off()
}
}
```

```{r coxph-datatables,echo=FALSE,results='asis'}
kable(summary.coxph.24b.gene, digits=c(0,0,9,9,9),caption = "Gene Level Tests for 24B-Gal4 Drivers", label = "tab:coxph-summary-24b")
kable(summary.coxph.24b.gene.gender, digits=c(0,0,9,9,9),caption = "Gene and Gender Level  Tests for 24B-Gal4 Drivers", label = "tab:coxph-summary-24b-gender")
```


```{r survival-24b-atg5}
    data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene=='Atg5'&data$Genotype!='Tm3/Tm6B',])
    survobj <- with(data.used, Surv(Age, Accidental==FALSE))
    survfit <- survfit(survobj~Genotype, data=data.used)
    plot(survfit, main = sprintf('Survival of Atg5 shRNA Flies with 24B-GAL4 Driver', gene),
         xlab= 'Days',
         ylab= 'Percent Survival',
         yscale=100, 
         las = 1,
         lty=c(1,2,2),
         col= genotype.colors)
    legend("bottomleft", levels(data.used$Genotype), col=genotype.colors, lty=c(1,2,2), bty="n")
```


```{r survival-24b-atg5-gender,fig=TRUE,echo=FALSE, dev=c('png','pdf')}
par(mfrow=c(1,2))
    data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene=='Atg5'&data$Genotype!='Tm3/Tm6B'&data$Gender=='male',])
    survobj <- with(data.used, Surv(Age, Accidental==FALSE))
    survfit <- survfit(survobj~Genotype, data=data.used)
    plot(survfit, main = 'Males',
         xlab= 'Days',
         ylab= 'Percent Survival',
         yscale=100, 
         las = 1,
         lty=c(1,2,2),
         col= genotype.colors)
    legend("bottomleft", levels(data.used$Genotype), col=genotype.colors, lty=c(1,2,2), bty="n")

    data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene=='Atg5'&data$Genotype!='Tm3/Tm6B'&data$Gender=='female',])
    survobj <- with(data.used, Surv(Age, Accidental==FALSE))
    survfit <- survfit(survobj~Genotype, data=data.used)
    plot(survfit, main = 'Females',
         xlab= 'Days',
         ylab= 'Percent Survival',
         yscale=100, 
         las = 1,
         lty=c(1,2,2),
         col= genotype.colors)
    legend("bottomleft", levels(data.used$Genotype), col=genotype.colors, lty=c(1,2,2), bty="n")
```

```{r survival-24b-atg8a}
    data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene=='Atg8a'&data$Genotype!='Tm3/Tm6B',])
    survobj <- with(data.used, Surv(Age, Accidental==FALSE))
    survfit <- survfit(survobj~Genotype, data=data.used)
    plot(survfit, main = sprintf('Survival of Atg8a shRNA Flies with 24B-GAL4 Driver', gene),
         xlab= 'Days',
         ylab= 'Percent Survival',
         yscale=100, 
         las = 1,
         lty=c(1,2,2),
         col= genotype.colors)
    legend("bottomleft", levels(data.used$Genotype), col=genotype.colors, lty=c(1,2,2), bty="n")
```

```{r survival-24b-atg8a-gender}
par(mfrow=c(1,2))
    data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene=='Atg8a'&data$Genotype!='Tm3/Tm6B'&data$Gender=='male',])
    survobj <- with(data.used, Surv(Age, Accidental==FALSE))
    survfit <- survfit(survobj~Genotype, data=data.used)
    plot(survfit, main = 'Males',
         xlab= 'Days',
         ylab= 'Percent Survival',
         yscale=100, 
         las = 1,
         lty=c(1,2,2),
         col= genotype.colors)
    legend("bottomleft", levels(data.used$Genotype), col=genotype.colors, lty=c(1,2,2), bty="n")

    data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene=='Atg8a'&data$Genotype!='Tm3/Tm6B'&data$Gender=='female',])
    survobj <- with(data.used, Surv(Age, Accidental==FALSE))
    survfit <- survfit(survobj~Genotype, data=data.used)
    plot(survfit, main = 'Females',
         xlab= 'Days',
         ylab= 'Percent Survival',
         yscale=100, 
         las = 1,
         lty=c(1,2,2),
         col= genotype.colors)
    legend("bottomleft", levels(data.used$Genotype), col=genotype.colors, lty=c(1,2,2), bty="n")
```

```{r survival-24b-atg8b}
    data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene=='Atg8b'&data$Genotype!='Tm3/Tm6B',])
    survobj <- with(data.used, Surv(Age, Accidental==FALSE))
    survfit <- survfit(survobj~Genotype, data=data.used)
    plot(survfit, main = sprintf('Survival of Atg8b shRNA Flies with 24B-GAL4 Driver', gene),
         xlab= 'Days',
         ylab= 'Percent Survival',
         yscale=100, 
         las = 1,
         lty=c(1,2,2),
         col= genotype.colors)
    legend("bottomleft", levels(data.used$Genotype), col=genotype.colors, lty=c(1,2,2), bty="n")
```

```{r survival-24b-atg8b-gender}
par(mfrow=c(1,2))
    data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene=='Atg8b'&data$Genotype!='Tm3/Tm6B'&data$Gender=='male',])
    survobj <- with(data.used, Surv(Age, Accidental==FALSE))
    survfit <- survfit(survobj~Genotype, data=data.used)
    plot(survfit, main = 'Males',
         xlab= 'Days',
         ylab= 'Percent Survival',
         yscale=100, 
         las = 1,
         lty=c(1,2,2),
         col= genotype.colors)
    legend("bottomleft", levels(data.used$Genotype), col=genotype.colors, lty=c(1,2,2), bty="n")

    data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene=='Atg8b'&data$Genotype!='Tm3/Tm6B'&data$Gender=='female',])
    survobj <- with(data.used, Surv(Age, Accidental==FALSE))
    survfit <- survfit(survobj~Genotype, data=data.used)
    plot(survfit, main = 'Females',
         xlab= 'Days',
         ylab= 'Percent Survival',
         yscale=100, 
         las = 1,
         lty=c(1,2,2),
         col= genotype.colors)
    legend("bottomleft", levels(data.used$Genotype), col=genotype.colors, lty=c(1,2,2), bty="n")
```

```{r survival-24b-HLH106}
    data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene=='Atg8b'&data$Genotype!='Tm3/Tm6B',])
    survobj <- with(data.used, Surv(Age, Accidental==FALSE))
    survfit <- survfit(survobj~Genotype, data=data.used)
    plot(survfit, main = sprintf('Survival of HLH106 shRNA Flies with 24B-GAL4 Driver', gene),
         xlab= 'Days',
         ylab= 'Percent Survival',
         yscale=100, 
         las = 1,
         lty=c(1,2,2),
         col= genotype.colors)
    legend("bottomleft", levels(data.used$Genotype), col=genotype.colors, lty=c(1,2,2), bty="n")
```

```{r survival-24b-HLH106-gender}
par(mfrow=c(1,2))
    data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene=='HLH106'&data$Genotype!='Tm3/Tm6B'&data$Gender=='male',])
    survobj <- with(data.used, Surv(Age, Accidental==FALSE))
    survfit <- survfit(survobj~Genotype, data=data.used)
    plot(survfit, main = 'Males',
         xlab= 'Days',
         ylab= 'Percent Survival',
         yscale=100, 
         las = 1,
         lty=c(1,2,2),
         col= genotype.colors)
    legend("bottomleft", levels(data.used$Genotype), col=genotype.colors, lty=c(1,2,2), bty="n")

    data.used <- droplevels(data[data$Driver=='24B-Gal4'&data$Gene=='HLH106'&data$Genotype!='Tm3/Tm6B'&data$Gender=='female',])
    survobj <- with(data.used, Surv(Age, Accidental==FALSE))
    survfit <- survfit(survobj~Genotype, data=data.used)
    plot(survfit, main = 'Females',
         xlab= 'Days',
         ylab= 'Percent Survival',
         yscale=100, 
         las = 1,
         lty=c(1,2,2),
         col= genotype.colors)
    legend("bottomleft", levels(data.used$Genotype), col=genotype.colors, lty=c(1,2,2), bty="n")
```


```{r coxph-datatables-atg}
atg5.summary <- rbind(coxph.table.24b.Atg5,coxph.table.24b.Atg5.female,coxph.table.24b.Atg5.male)
rownames(atg5.summary) <- c("Total", "Male", "Female")
kable(atg5.summary, digits=c(0,3,3,3,8),caption = "Gene Level Cox Proportional Hazard Tests for 24B-Gal4 Drivers and Atg5 Knockdown", label = "tab:coxph.table.24b.atg5")

atg8a.summary <- rbind(coxph.table.24b.Atg8a,coxph.table.24b.Atg8a.female,coxph.table.24b.Atg8a.male)
rownames(atg8a.summary) <- c("Total", "Male", "Female")
kable(atg8a.summary, digits=c(0,3,3,3,3),caption = "Gene Level Cox Proportional Hazard Tests for 24B-Gal4 Drivers and Atg8a Knockdown", label = "tab:coxph.table.24b.atg8a")

atg8b.summary <- rbind(coxph.table.24b.Atg8b,coxph.table.24b.Atg8b.female,coxph.table.24b.Atg8b.male)
rownames(atg8b.summary) <- c("Total", "Male", "Female")
kable(atg8b.summary, digits=c(0,3,3,3,15),caption = "Gene Level Cox Proportional Hazard Tests for 24B-Gal4 Drivers and Atg8b Knockdown", label = "tab:coxph.table.24b.atg8b")

HLH106.summary <- rbind(coxph.table.24b.HLH106,coxph.table.24b.HLH106.female,coxph.table.24b.HLH106.male)
rownames(HLH106.summary) <- c("Total", "Male", "Female")
kable(HLH106.summary, digits=c(0,3,3,3,15),caption = "Gene Level Cox Proportional Hazard Tests for 24B-Gal4 Drivers and HLH106 Knockdown", label = "tab:coxph.table.24b.HLH106")
```

## Atg5 Knockdown

The summary statistics from this analysis are shown in the Table below:

\ref{tab:coxph.table.24b.atg5}.  

In the case of muscle *Atg5* knockdown flies, we observed the knockdown flies dying later than the wildtypes with a hazard ratio of `r atg5.summary['Total',]$Hazard.Ratio` (p=`r atg5.summary['Total',]$p`).  This was true for both Males (HR=`r atg5.summary['Male',]$Hazard.Ratio`, p=`r atg5.summary['Male',]$p`) and Females (HR=`r atg5.summary['Female',]$Hazard.Ratio`, p=`r atg5.summary['Female',]$p`).


## Atg8a Knockdown

The summary statistics from this analysis are shown in the table below:

Table \ref{tab:coxph.table.24b.atg8a}.  

In the case of muscle *Atg8b* knockdown flies, we observed no significant differences with a hazard ratio of `r atg8a.summary['Total',]$Hazard.Ratio` (p=`r atg8a.summary['Total',]$p`).  This was true for both Males (HR=`r atg8a.summary['Male',]$Hazard.Ratio`, p=`r atg8a.summary['Male',]$p`) and Females (HR=`r atg8a.summary['Female',]$Hazard.Ratio`, p=`r atg8a.summary['Female',]$p`).

## Atg8b Knockdown

The summary statistics from this analysis are shown in the Table below


\ref{tab:coxph.table.24b.atg8b}.  

In the case of muscle *Atg8b* knockdown flies, we observed the knockdown flies dying later than the wildtypes with a hazard ratio of `r atg8b.summary['Total',]$Hazard.Ratio` (p=`r atg8b.summary['Total',]$p`).  This was true for both Males (HR=`r atg8b.summary['Male',]$Hazard.Ratio`, p=`r atg8b.summary['Male',]$p`) and Females (HR=`r atg8b.summary['Female',]$Hazard.Ratio`, p=`r atg8b.summary['Female',]$p`).

## HLH106 Knockdown

The summary statistics from this analysis are shown in the Table below

\ref{tab:coxph.table.24b.HLH106}.  

In the case of muscle *HLH106* knockdown flies, we observed the knockdown flies dying later than the wildtypes with a hazard ratio of `r HLH106.summary['Total',]$Hazard.Ratio` (p=`r HLH106.summary['Total',]$p`).  This was true for both Males (HR=`r HLH106.summary['Male',]$Hazard.Ratio`, p=`r HLH106.summary['Male',]$p`) and Females (HR=`r HLH106.summary['Female',]$Hazard.Ratio`, p=`r HLH106.summary['Female',]$p`).


## Session Information

```{r sessionInfo}
sessionInfo()
```