Analysis of Birth Ratios
==========================

Experimental Design
--------------------

```{r data-input, echo=FALSE}
spreadsheet_location <- 'https://docs.google.com/spreadsheet/pub?key=0Aitfmor6rCjYdFZsdGpxcmhLX2lNd2lNWUN6T014Qnc&single=true&gid=0&output=csv' 

mapping.data <- read.csv('../Data/gene_mapping.csv')
data_file_name <- '../Data/birth_worksheet.csv'
download.file(spreadsheet_location, data_file_name, method="curl")
data <- read.csv(data_file_name)
data$Total <- rowSums(data[,10:13])
all.flies <- colSums(data[,10:13])
require(plyr, quietly=T)
data$Cross <- as.factor(paste(data$Male, data$Female, sep=" x "))
data$Driver <- as.factor(sapply(strsplit(as.character(data$Female), "/"),"[", 1))
#releveled UAS showing TSC1 then Raptor
data$UAS <- as.factor(sapply(strsplit(as.character(data$Male), "/"),"[", 1))
data$UAS <- factor(data$UAS, levels=c('31039','31314','35144','31528','31529','34814'))
#merged with gene mapping file
data <- merge(mapping.data, data, by="UAS")
#mark crosses as cold
data$temp18c <- substring(data$CrossName, 1,2) == "x-"
data.18c <- data[data$temp18c==TRUE,]
data.c179 <- data[data$Driver=='C179-Gal4',]
data <- droplevels(data[data$temp18c==FALSE&data$Driver!='C179-Gal4',])
```

These data are stored with the raw data saved in`r data_file_name`.  This script was most recently run on `r date()`.

Strain Summaries
------------------

```{r data-analysis, echo=FALSE}
gene.summarised.data <- ddply(data, .(Driver,Gene), summarise, 
      KD = sum(WT),
      shRNA = sum(Ser.Sb),
      GAL4 = sum(Tm6B),
      WT = sum(Tm6B.Ser.Sb),
      Total = as.integer(sum(Total)))

gene.summarised.data$pval <-rep(NA, dim(gene.summarised.data)[1])
for (n in seq(1,dim(gene.summarised.data)[1])) {   
  values <- gene.summarised.data[n,3:5]
  test <- chisq.test(values, p=rep(1,3), rescale.p=T)
  gene.summarised.data[n,]$pval <- test$p.value  
}
gene.summarised.data$padj <- p.adjust(gene.summarised.data$pval, method="BH")

driver.summarised.data <- ddply(data, .(Driver,UAS), summarise, 
      KD = sum(WT),
      shRNA = sum(Ser.Sb),
      GAL4 = sum(Tm6B),
      WT = sum(Tm6B.Ser.Sb),
      Total = as.integer(sum(Total)))

driver.summarised.data$pval <-rep(NA, dim(driver.summarised.data)[1])
for (n in seq(1,dim(driver.summarised.data)[1])) {   
  values <- driver.summarised.data[n,3:5]
  test <- chisq.test(values, p=rep(1,3), rescale.p=T)
  driver.summarised.data[n,]$pval <- test$p.value  
}
driver.summarised.data$padj <- p.adjust(driver.summarised.data$pval, method="BH")

summarised.data <- ddply(data, ~Cross, summarise, 
      KD = sum(WT),
      shRNA = sum(Ser.Sb),
      GAL4 = sum(Tm6B),
      WT = sum(Tm6B.Ser.Sb),
      Total = as.integer(sum(Total)))
all.progeny <-rep(NA, length(levels(data$Cross)))
names(all.progeny) <- levels(data$Cross)
for (n in seq(1,length(levels(data$Cross)))){
  values <- summarised.data[n,2:5]
  test <- chisq.test(values, p=rep(1,4), rescale.p=T)
  all.progeny[levels(data$Cross)[n]] <- test$p.value  
}
all.progeny.padj <- cbind(pval=all.progeny, padj=p.adjust(all.progeny, method="BH"))

summarised.data$pval <-rep(NA, dim(summarised.data)[1])
for (n in seq(1,length(levels(summarised.data$Cross)))) {
  values <- summarised.data[n,2:4]
  test <- chisq.test(values, p=rep(1,3), rescale.p=T)
  summarised.data[n,]$pval <- test$p.value  
}
summarised.data$padj <- p.adjust(summarised.data$pval, method="BH")

summarised.data.gendered <- ddply(data, .(UAS, Driver,Gender), summarise, 
      KD = sum(WT),
      shRNA = sum(Ser.Sb),
      GAL4 = sum(Tm6B),
      WT = sum(Tm6B.Ser.Sb),
      Total = as.integer(sum(Total)))

summarised.data.gendered$pval <-rep(NA, dim(summarised.data.gendered)[1])
for (n in seq(1,dim(summarised.data.gendered)[1])) {   
  values <- summarised.data.gendered[n,4:6]
  test <- chisq.test(values, p=rep(1,3), rescale.p=T)
  summarised.data.gendered[n,]$pval <- test$p.value  
}
summarised.data.gendered$padj <- p.adjust(summarised.data.gendered$pval, method="BH")

driver.summarised.data.c179 <- ddply(data.c179, .(Driver,UAS), summarise, 
      KD = sum(WT),
      shRNA = sum(Ser.Sb),
      GAL4 = sum(Tm6B),
      WT = sum(Tm6B.Ser.Sb),
      Total = as.integer(sum(Total)))

driver.summarised.data.c179$pval <-rep(NA, dim(driver.summarised.data.c179)[1])
for (n in seq(1,dim(driver.summarised.data.c179)[1])) {   
  values <- driver.summarised.data.c179[n,3:5]
  test <- chisq.test(values[c(1,3)], p=c(1,1), rescale.p=T)
  driver.summarised.data.c179[n,]$pval <- test$p.value  
}
driver.summarised.data.c179$padj <- p.adjust(driver.summarised.data.c179$pval, method="BH")

#for gendered 179 results
summarised.data.gendered.c179 <- ddply(data.c179, .(UAS, Driver,Gender), summarise, 
      KD = sum(WT),
      shRNA = sum(Ser.Sb),
      GAL4 = sum(Tm6B),
      WT = sum(Tm6B.Ser.Sb),
      Total = as.integer(sum(Total)))
summarised.data.gendered.c179 <- droplevels(summarised.data.gendered.c179)

summarised.data.gendered.c179$pval <-rep(NA, dim(summarised.data.gendered.c179)[1])
for (n in seq(1,dim(summarised.data.gendered.c179)[1])) {   
  values <- summarised.data.gendered.c179[n,4:6]
  test <- chisq.test(values[c(1,3)], p=rep(1,2), rescale.p=T)
  summarised.data.gendered.c179[n,]$pval <- test$p.value  
}
summarised.data.gendered.c179$padj <- p.adjust(summarised.data.gendered.c179$pval, method="BH")


driver.summarised.data.18c <- ddply(data.18c, .(Driver,UAS), summarise, 
      KD = sum(WT),
      shRNA = sum(Ser.Sb),
      GAL4 = sum(Tm6B),
      WT = sum(Tm6B.Ser.Sb),
      Total = as.integer(sum(Total)))

driver.summarised.data.18c$pval <-rep(NA, dim(driver.summarised.data.18c)[1])
for (n in seq(1,dim(driver.summarised.data.18c)[1])) {   
  values <- driver.summarised.data.18c[n,3:5]
  test <- chisq.test(values, p=rep(1,3), rescale.p=T)
  driver.summarised.data.18c[n,]$pval <- test$p.value  
}
driver.summarised.data.18c$padj <- p.adjust(driver.summarised.data.18c$pval, method="BH")
```

So far this analysis includes `r sum(all.flies)` flies from `r length(levels(data$Cross))` crosses.  Statistics are based on a Chi Squared Test, and are  adjusted for multiple observation by the method of Benjamini and Hochberg.  These statistics exclude the genotypically WT (dual balancer chromosome) strains, which have significantly less progeny in nearly all cases (see Table below):

```{r all-progeny-tables,echo=FALSE,results='asis', message=FALSE}
require(xtable)
print(xtable(all.progeny.padj, caption="Summarized Data Including Double Balancer Strains.  The values are all significant because including the double balancer strain is sublethal in every case", label="tab:all_progeny_data", digits = c(0,6,6)), type='html')
```

After excluding the double balancer strains from the analysis, counts and p-values are shown in the table and figures below for all genes:

```{r gene-summarized-data,echo=FALSE,results='asis'}
print(xtable(gene.summarised.data, caption="Gene Level Summarised Data", label="tab:gene_summarised_data", digits = c(rep(0,8),8,8)), include.rownames=F, type='html')
```

And summarized at the level of the shRNA clone

```{r uas-summarized-data,echo=FALSE,results='asis', warning=FALSE}
print(xtable(driver.summarised.data[order(driver.summarised.data$Driver,driver.summarised.data$UAS),], caption="Driver Level Summarised Data", label="tab:driver_summarised_data", digits = c(rep(0,8),5,5)), include.rownames=F, type='html')
```

## 24B-GAL4 Mediated Knockdown

We observed nearly no flies born from this cross, either from males or females.

```{r 24b-births-all, echo=FALSE, dev=c('png','pdf'), warning=FALSE}
driver.summarised.data$Total.No.WT <- rowSums(driver.summarised.data[,c('KD','shRNA','GAL4') ])
driver.summarised.data$KD.prop <- driver.summarised.data$KD/driver.summarised.data$Total.No.WT
driver.summarised.data$shRNA.prop <- driver.summarised.data$shRNA/driver.summarised.data$Total.No.WT
driver.summarised.data$GAL4.prop <- driver.summarised.data$GAL4/driver.summarised.data$Total.No.WT

#calculated standard errors
driver.summarised.data$KD.err <- with(driver.summarised.data, sqrt((KD.prop*(1-KD.prop)/Total.No.WT)))
driver.summarised.data$shRNA.err <- with(driver.summarised.data, sqrt((shRNA.prop*(1-shRNA.prop)/Total.No.WT)))
driver.summarised.data$GAL4.err <- with(driver.summarised.data, sqrt((GAL4.prop*(1-GAL4.prop)/Total.No.WT)))
driver.summarised.data <- driver.summarised.data[order(driver.summarised.data$KD.prop),]


require(RColorBrewer, quietly=T)
color.scheme <- brewer.pal(3,"Set1")
plot <- barplot(t(as.matrix(driver.summarised.data[driver.summarised.data$Driver=="24B-Gal4",c('KD.prop','shRNA.prop','GAL4.prop')])),
        names.arg=driver.summarised.data[driver.summarised.data$Driver=="24B-Gal4",]$UAS,
        beside=T,las=2,
        ylab="Proportion of Eclosed Flies",
        ylim=c(0,0.65),
        col=color.scheme,
        main="24B-Gal4 Driver")
abline(a=1/3,b=0, lty=2)
abline(v=12.5)
legend("topright", c("shRNA/GAL4", "shRNA/+", "GAL4/+"), bty="n", fill=color.scheme)

superpose.eb <- 
 function (x, y, ebl, ebu = ebl, length = 0.08, ...) 
 arrows(x, y + ebu, x, y - ebl, angle = 90, code = 3, 
 length = length, ...)

superpose.eb(plot, 
             t(as.matrix(driver.summarised.data[driver.summarised.data$Driver=="24B-Gal4",c('KD.prop','shRNA.prop','GAL4.prop')])),
             t(as.matrix(driver.summarised.data[driver.summarised.data$Driver=="24B-Gal4",c('KD.err','shRNA.err','GAL4.err')])))
```

```{r 24b-births-gendered, echo=FALSE, dev=c('png','pdf'), warning=FALSE}
summarised.data.gendered$Total.No.WT <- rowSums(summarised.data.gendered[,c('KD','shRNA','GAL4') ])
summarised.data.gendered$KD.prop <- summarised.data.gendered$KD/summarised.data.gendered$Total.No.WT
summarised.data.gendered$shRNA.prop <- summarised.data.gendered$shRNA/summarised.data.gendered$Total.No.WT
summarised.data.gendered$GAL4.prop <- summarised.data.gendered$GAL4/summarised.data.gendered$Total.No.WT

#calculated standard errors
summarised.data.gendered$KD.err <- with(summarised.data.gendered, sqrt((KD.prop*(1-KD.prop)/Total.No.WT)))
summarised.data.gendered$shRNA.err <- with(summarised.data.gendered, sqrt((shRNA.prop*(1-shRNA.prop)/Total.No.WT)))
summarised.data.gendered$GAL4.err <- with(summarised.data.gendered, sqrt((GAL4.prop*(1-GAL4.prop)/Total.No.WT)))
summarised.data.gendered <- summarised.data.gendered[order(summarised.data.gendered$KD.prop),]

par(mfrow=c(2,1))
for (gender in levels(summarised.data.gendered$Gender)) {
color.scheme <- brewer.pal(3,"Set1")
plot <- barplot(t(as.matrix(summarised.data.gendered[summarised.data.gendered$Driver=="24B-Gal4"&summarised.data.gendered$Gender==gender,c('KD.prop','shRNA.prop','GAL4.prop')])),
        names.arg=summarised.data.gendered[summarised.data.gendered$Driver=="24B-Gal4"&summarised.data.gendered$Gender==gender,]$UAS,
        beside=T,las=2,
        ylab="Proportion of Eclosed Flies",
        ylim=c(0,0.65),
        col=color.scheme,
        main=paste("24B-Gal4 Driver", gender, sep="--"))
abline(a=1/3,b=0, lty=2)
abline(v=12.5)
legend("topright", c("shRNA/GAL4", "shRNA/+", "GAL4/+"), bty="n", fill=color.scheme)

superpose.eb(plot, 
             t(as.matrix(summarised.data.gendered[summarised.data.gendered$Driver=="24B-Gal4"&summarised.data.gendered$Gender==gender,c('KD.prop','shRNA.prop','GAL4.prop')])),
             t(as.matrix(summarised.data.gendered[summarised.data.gendered$Driver=="24B-Gal4"&summarised.data.gendered$Gender==gender,c('KD.err','shRNA.err','GAL4.err')])))
}

```

### Effects of 18C
Since we observed few births at 24C, we repeated several crosses at 18C using the TSC shRNA lines.  These data are summarised in Table and Figure below:

```{r data-tables-18c,echo=FALSE,results='asis'}
print(xtable(driver.summarised.data.18c, caption="Driver Level Summarized Data for 18C Crosses", label="tab:18c-data", digits = c(rep(0,8),7,7)), include.rownames=F, type='html')
```

```{r 24b-births-18c, echo=FALSE, dev=c('png','pdf'), warning=FALSE}
driver.summarised.data.18c$Total.No.WT <- rowSums(driver.summarised.data.18c[,c('KD','shRNA','GAL4') ])
driver.summarised.data.18c$KD.prop <- driver.summarised.data.18c$KD/driver.summarised.data.18c$Total.No.WT
driver.summarised.data.18c$shRNA.prop <- driver.summarised.data.18c$shRNA/driver.summarised.data.18c$Total.No.WT
driver.summarised.data.18c$GAL4.prop <- driver.summarised.data.18c$GAL4/driver.summarised.data.18c$Total.No.WT

#calculated standard errors
driver.summarised.data.18c$KD.err <- with(driver.summarised.data.18c, sqrt((KD.prop*(1-KD.prop)/Total.No.WT)))
driver.summarised.data.18c$shRNA.err <- with(driver.summarised.data.18c, sqrt((shRNA.prop*(1-shRNA.prop)/Total.No.WT)))
driver.summarised.data.18c$GAL4.err <- with(driver.summarised.data.18c, sqrt((GAL4.prop*(1-GAL4.prop)/Total.No.WT)))

plot <- barplot(t(as.matrix(driver.summarised.data.18c[,c('KD.prop','shRNA.prop','GAL4.prop')])),
        names.arg=driver.summarised.data.18c[driver.summarised.data.18c$Driver=="24B-Gal4",]$UAS,
        beside=T,las=2,
        ylab="Proportion of Eclosed Flies",
        ylim=c(0,0.9),
        col=color.scheme,
        main="24B-Gal4 Driver at 18C")
abline(a=1/3,b=0, lty=2)
legend("topright", c("shRNA/GAL4", "shRNA/+", "GAL4/+"), bty="n", fill=color.scheme)


superpose.eb(plot, 
             t(as.matrix(driver.summarised.data.18c[,c('KD.prop','shRNA.prop','GAL4.prop')])),
             t(as.matrix(driver.summarised.data.18c[,c('KD.err','shRNA.err','GAL4.err')])))
```

## Hand-GAL4 Mediated Knockdown

Hand-GAL4 no obvious deficiencies in the number of flies born:

```{r hand-births-all, echo=FALSE, dev=c('png','pdf')}
plot <- barplot(t(as.matrix(driver.summarised.data[driver.summarised.data$Driver=="Hand-Gal4",c('KD.prop','shRNA.prop','GAL4.prop')])),
        names.arg=driver.summarised.data[driver.summarised.data$Driver=="Hand-Gal4",]$UAS,
        beside=T,las=2,
        ylab="Proportion of Eclosed Flies",
        ylim=c(0,0.6),
        col=color.scheme,
        main="Hand-Gal4 Driver")
abline(a=1/3,b=0, lty=2)
abline(v=12.5)
legend("topright", c("shRNA/GAL4", "shRNA/+", "GAL4/+"), bty="n", fill=color.scheme)


superpose.eb(plot, 
             t(as.matrix(driver.summarised.data[driver.summarised.data$Driver=="Hand-Gal4",c('KD.prop','shRNA.prop','GAL4.prop')])),
             t(as.matrix(driver.summarised.data[driver.summarised.data$Driver=="Hand-Gal4",c('KD.err','shRNA.err','GAL4.err')])))
```

And separated out by gender:

```{r hand-births-gendered, echo=FALSE, dev=c('png','pdf')}
par(mfrow=c(2,1))
for (gender in levels(summarised.data.gendered$Gender)) {
color.scheme <- brewer.pal(3,"Set1")
plot <- barplot(t(as.matrix(summarised.data.gendered[summarised.data.gendered$Driver=="Hand-Gal4"&summarised.data.gendered$Gender==gender,c('KD.prop','shRNA.prop','GAL4.prop')])),
        names.arg=summarised.data.gendered[summarised.data.gendered$Driver=="Hand-Gal4"&summarised.data.gendered$Gender==gender,]$UAS,
        beside=T,las=2,
        ylab="Proportion of Eclosed Flies",
        ylim=c(0,0.65),
        col=color.scheme,
        main=paste("Hand-Gal4 Driver", gender, sep="--"))
abline(a=1/3,b=0, lty=2)
abline(v=12.5)
legend("topright", c("shRNA/GAL4", "shRNA/+", "GAL4/+"), bty="n", fill=color.scheme)

superpose.eb(plot, 
             t(as.matrix(summarised.data.gendered[summarised.data.gendered$Driver=="Hand-Gal4"&summarised.data.gendered$Gender==gender,c('KD.prop','shRNA.prop','GAL4.prop')])),
             t(as.matrix(summarised.data.gendered[summarised.data.gendered$Driver=="Hand-Gal4"&summarised.data.gendered$Gender==gender,c('KD.err','shRNA.err','GAL4.err')])))
}
```


## C179 Driven Knockdown

There was also complete lethality observed for C179 driven knockdowns.  Below is the data shown at the shRNA and shRNA/gendered levels

```{r data-tables-c179,echo=FALSE,results='asis'}
print(xtable(driver.summarised.data.c179, caption="Driver Level Summarized Data for C179-Gal4 Crosses", label="tab:179-all", digits = c(rep(0,8),3,3)), include.rownames=F, type='html')
print(xtable(summarised.data.gendered.c179, caption="Driver Level, Gender Separated Summarized Data for C179-Gal4 Crosses", label="tab:179-gendered", digits = c(rep(0,9),4,4)), include.rownames=F, type='html')
```

```{r c179-births-all, echo=FALSE, dev=c('png','pdf')}
driver.summarised.data.c179$Total.No.WT <- rowSums(driver.summarised.data.c179[,c('KD','shRNA','GAL4') ])
driver.summarised.data.c179$KD.prop <- driver.summarised.data.c179$KD/driver.summarised.data.c179$Total.No.WT
driver.summarised.data.c179$GAL4.prop <- driver.summarised.data.c179$GAL4/driver.summarised.data.c179$Total.No.WT

#calculated standard errors
driver.summarised.data.c179$KD.err <- with(driver.summarised.data.c179, sqrt((KD.prop*(1-KD.prop)/Total.No.WT)))
driver.summarised.data.c179$GAL4.err <- with(driver.summarised.data.c179, sqrt((GAL4.prop*(1-GAL4.prop)/Total.No.WT)))

plot <- barplot(t(as.matrix(driver.summarised.data.c179[,c('KD.prop','GAL4.prop')])),
        names.arg=driver.summarised.data.c179$UAS,
        beside=T,las=2,
        ylab="Proportion of Eclosed Flies",
        ylim=c(0,0.9),
        col=color.scheme[c(1,3)],
        main="C179 Driver")
abline(a=1/2,b=0, lty=2)
legend("topleft", c("shRNA/GAL4","GAL4/+"), bty="n", fill=color.scheme[c(1,3)])


superpose.eb(plot, 
             t(as.matrix(driver.summarised.data.c179[,c('KD.prop','GAL4.prop')])),
             t(as.matrix(driver.summarised.data.c179[,c('KD.err','GAL4.err')])))
```

The differences in lethality between genders was only present for clone 31529.  The p-value for this was `r fisher.test(rbind(
      subset(summarised.data.gendered.c179, Gender == "Female"&UAS==31529)[c('KD','GAL4')],
      subset(summarised.data.gendered.c179, Gender == "Male"&UAS==31529)[c('KD','GAL4')]
                  ))$p.value`.  This is shown below:


```{r c179-gendered-figure, echo=FALSE, dev=c('png','pdf')}
summarised.data.gendered.c179$Total.No.WT <- rowSums(summarised.data.gendered.c179[,c('KD','GAL4') ])
summarised.data.gendered.c179$KD.prop <- summarised.data.gendered.c179$KD/summarised.data.gendered.c179$Total.No.WT
summarised.data.gendered.c179$GAL4.prop <- summarised.data.gendered.c179$GAL4/summarised.data.gendered.c179$Total.No.WT

#calculated standard errors
summarised.data.gendered.c179$KD.err <- with(summarised.data.gendered.c179, sqrt((KD.prop*(1-KD.prop)/Total.No.WT)))
summarised.data.gendered.c179$GAL4.err <- with(summarised.data.gendered.c179, sqrt((GAL4.prop*(1-GAL4.prop)/Total.No.WT)))

require(reshape2)
plot <- barplot(as.matrix(dcast(summarised.data.gendered.c179, Gender~UAS, value.var='KD.prop')[2:4]),
        names.arg=levels(summarised.data.gendered.c179$UAS),
        beside=T,las=2,
        ylab="Proportion of Knockdown Flies Eclosed",
        ylim=c(0,0.5),
        main="C157-Gal4 Driver By Gender")
abline(a=1/2,b=0, lty=2)
legend("topright", levels(summarised.data.gendered.c179$Gender), bty="n", fill=grey.colors(2))


superpose.eb(plot, 
             as.matrix(dcast(summarised.data.gendered.c179, Gender~UAS, value.var='KD.prop')[2:4]),
             as.matrix(dcast(summarised.data.gendered.c179, Gender~UAS, value.var='KD.err')[2:4]))
```

`
## General Traits

### Fecundity over time

```{r generations, echo=FALSE, dev=c('png','pdf')}
with(data, plot(jitter(Generation),Total,
                pch=18,las=2,
                ylab="Flies Born Per Generation",
                xlab="Generation Number"))
generation.lm <- lm(Total~Generation,data=data)
lines(c(1:7), predict(generation.lm, newdata=list(Generation=c(1:7))))
lines(c(1:7), 
      predict(generation.lm, newdata=list(Generation=c(1:7)), interval="confidence")[,'lwr'], 
      lty=2)
lines(c(1:7), 
      predict(generation.lm, newdata=list(Generation=c(1:7)), interval="confidence")[,'upr'], 
      lty=2)
```

To test how many generations we can get out of 10 female flies in a vial, we examined the number of births in a vial, vs the generation of that vial.  These data are in the figure above.  There is no detectable decrease in fecundity over time (p=`r anova(generation.lm, drop1(generation.lm))[1,5]`).

### Gender Differences

On the other hand, there was a significant difference between the number of Male flies born vs the number of Female flies born (p=`r t.test(data$Total~data$Gender)$p.value`).  This is visulized in the figure below:

```{r gender, echo=FALSE, dev=c('png','pdf')}
boxplot(Total~Gender,data=data, col="grey",
        ylab="Flies Eclosed Per Vial")
```
     

## Session Information
```{r sessionInfo, echo=F}
sessionInfo()
```